src/main.rs

```diff
@@ -109,6 +172,33 @@ fn main() -> Result<()> {
     let mut buffer = Vec::new();
     input.read_to_end(&mut buffer)?;
 
+    let schema_string_read_result = opts.read_schema_to_string();
+    let query_string_read_result = opts.read_query_to_string();
+    let input_json: serde_json::Value = serde_json::from_slice(&buffer)?;
+    let mut scale_factor = 1.0;
+
+    if let (Ok(schema_string), Ok(query_string)) =
+        (schema_string_read_result, query_string_read_result)
+    {
+        let document_definition =
+            BluejaySchemaAnalyzer::create_definition_document(&schema_string).unwrap();
+
+        let schema_result = BluejaySchemaAnalyzer::create_schema_definition(&document_definition);
+
+        scale_factor = match schema_result {
+            Ok(schema) => {
+                BluejaySchemaAnalyzer::analyze_schema_definition(schema, &query_string, &input_json)
+                    .unwrap_or(1.0)
```

Comment:
If there's an error analyzing the schema definition, should that error be output somewhere?

Reply:
(also related to other comment) I went with outputting the error message and continuing as normal. i used the anyhow error you suggested. One thing I did was us the `unwrap_or_else` when we call `analyze_schema_definition`. Im not sure, but think that is how we can handle the anyhow errors and set the defaults. (vs using a `?`).